cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -48,10 +48,6 @@ if (DEFINED ENV{THRIFT_HOME})
   set(THRIFT_HOME "$ENV{THRIFT_HOME}")
 endif()
 
-if (DEFINED ENV{ARROW_HOME})
-  set(ARROW_HOME "$ENV{ARROW_HOME}")
-endif()
```

Comment:
Moving this to FindArrow.cmake was surprising, but I'm not sure if there's a better way